### PR TITLE
Update to use stable Viem / Wagmi versions

### PR DESCRIPTION
**Pull request by zencephalon**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,11 +100,11 @@
...
   "peerDependencies": {
     "@tanstack/react-query": ">=5.0.0",
-    "@wagmi/core": "beta",
+    "@wagmi/core": "^2.6.5",
     "op-viem": "1.3.1-alpha",
     "typescript": ">=5.0.4",
-    "viem": "beta",
-    "wagmi": "beta"
+    "viem": "^2.7.9",
+    "wagmi": "^2.5.7"
```
Comment (lukasrosario) [inline package.json:107]: i think these be `2.x`? that's what we have in op-viem and how wagmi has their peer dependencies set up

**Comment by lukasrosario** [comment]
@zencephalon looking good! can we update the example app to use this version? do you think we can get the config changes in this update too? specifically, users shouldnt have to configure op chains as specified [here](https://www.opwagmi.sh/docs/configuration.html) anymore because viem 2 chains have op stack addresses.

**Comment by zencephalon** [comment]
@lukasrosario The Viem 2 actions don't map cleanly to op-viem or I would have done it already :/ E.g. Viem doesn't provide any simulation actions

File: package.json (hunk 1 above)
Comment (zencephalon) [reply package.json:107]: I can change it to that, my understanding of the `^` operator is that `^2.7.9` and `^2.x` have the same effect as they both allow any minor and patch version within 2

**Comment by zencephalon** [comment]
> @zencephalon looking good! can we update the example app to use this version?

The example app automatically uses whatever version is in the Workspace, no?

**Comment by lukasrosario** [comment]
ah i see ok, ack on the viem 2 actions. and yeah you're right it does use workspace, should be good. 
so only requests are:
- set up the peer dependencies so they're the same as op-viem, wagmi etc
- remove op-config & update docs
- `pnpm changeset` so we get an npm release pr

File: example/app/providers.tsx [hunk 2]
```diff
@@ -2,16 +2,16 @@
 
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
 import { createConfig, http, WagmiProvider } from 'wagmi'
-import { baseGoerli, goerli, optimismGoerli } from 'wagmi/chains'
+import { base, mainnet, optimism } from 'wagmi/chains'
```
Comment (lukasrosario) [inline example/app/providers.tsx:5]: let's use sepolia here

File: src/hooks/L1/useWriteDepositERC20.ts [hunk 3]
```diff
@@ -41,27 +41,25 @@ export type UseWriteDepositERC20ReturnType<config extends Config = OpConfig, con
...
     options,
   ) => {
-    const l2Chain = config.l2chains[l2ChainId]
-
-    if (!l2Chain) {
-      throw new Error('L2 chain not configured')
-    }
+    const { l2Chain, l1ChainId } = validateL2Chain(config, l2ChainId)
+    const l1StandardBridge = validatel1StandardBridgeContract(l1ChainId, l2Chain).address
+    console.log('l1StandardBridge', l1StandardBridge)
```
Comment (lukasrosario) [inline src/hooks/L1/useWriteDepositERC20.ts:57]: let's remove this

File: src/hooks/L1/useWriteDepositERC20.ts [hunk 4]
```diff
@@ -82,15 +80,13 @@ export function useWriteDepositERC20<config extends Config = OpConfig, context =
...
```
Comment (lukasrosario) [inline src/hooks/L1/useWriteDepositERC20.ts:85]: remove

**Comment by lukasrosario** [comment]
nice, can you run `pnpm changeset` so we get the release?

**Review by lukasrosario — APPROVED** [review]
🚀🚀 awesome. can follow up but looks like something happened w/ the test coverage and now it's reporting on the example app

**Pull request merged**